pvlib/test/test_irradiance.py

```diff
@@ -365,3 +368,29 @@ def test_erbs_all_scalar():
 
     for k, v in out.items():
         assert_allclose(v, expected[k], 5)
+
+
+def test_dirindex():
+    clearsky_data = tus.get_clearsky(times, model='ineichen',
+                                     linke_turbidity=3)
+    ghi = pd.Series([0, 0, 1038.62, 254.53], index=times)
+    ghi_clearsky = clearsky_data['ghi']
+    dni_clearsky = clearsky_data['dni']
+    zenith = ephem_data['apparent_zenith']
+    pressure = 93193.
+    tdew = 10.
+    out = irradiance.dirindex(ghi, ghi_clearsky, dni_clearsky,
+                              zenith, times, pressure=pressure,
+                              temp_dew=tdew)
+    dirint_close_values = irradiance.dirint(ghi, zenith, times,
+                                            pressure=pressure,
+                                            use_delta_kt_prime=True,
+                                            temp_dew=tdew).values
+    expected_out = np.array([np.nan, 0., 748.31562753, 630.72592644])
```

Comment:
Does this nan arise from 0/0? If so, I think most users would probably prefer to get a 0 here instead. Something like ``dni_dirindex[(dni_dirindex < 0) | dni_dirindex.isnan()] = 0.`` should work.